src/routes/accounts.py

```
@@ -18,8 +18,264 @@
+        await db.refresh(new_user)
+
+    except Exception:
+        await db.rollback()
+        raise HTTPException(status_code=500, detail="An error occurred during user creation.")
+
+    return UserRegistrationResponseSchema.model_validate(new_user)
+
+
+@router.post(
+    "/activate/",
+    response_model=MessageResponseSchema,
+    responses={
+        400: {
+            "model": AccountsErrorSchema,
+            "description": "Invalid or expired activation token, or user account is already active."
+        }
+    }
+)
+async def user_account_activation(data: UserActivationRequestSchema, db: AsyncSession = Depends(get_db)):
+    db_user = await get_user_by_email(db, data.email)
+
+    if db_user.is_active:
+        raise HTTPException(status_code=400, detail="User account is already active.")
+
+    stmt_token = select(ActivationTokenModel).where(ActivationTokenModel.user_id == db_user.id)
+    db_token = await db.scalar(stmt_token)
+
+    if (
+            not db_token
+            or db_token.token != data.token
+            or db_token.expires_at.replace(tzinfo=timezone.utc) < datetime.now(timezone.utc)
+    ):
+        raise HTTPException(status_code=400, detail="Invalid or expired activation token.")
+
+    db_user.is_active = True
+    await db.delete(db_token)
+    await db.commit()
+
+    return MessageResponseSchema.model_validate({"message": "User account activated successfully."})
+
+
+@router.post(
+    "/password-reset/request/",
+    response_model=MessageResponseSchema,
+)
+async def password_reset_token_request(data: PasswordResetRequestSchema, db: AsyncSession = Depends(get_db)):
+    db_user = await get_user_by_email(db, data.email)
+
+    if db_user and db_user.is_active:
+
+        stmt_delete_token = delete(PasswordResetTokenModel).where(PasswordResetTokenModel.user_id == db_user.id)
+        await db.execute(stmt_delete_token)
+
+        token = PasswordResetTokenModel(user=db_user)
+
+        db.add(token)
+        await db.commit()
```

Important: the password reset token request block performs DB deletes and token creation/commit but does not guard against DB exceptions. If a DB error occurs here the exception will propagate as a 500 instead of returning the generic 200 message (the requirement explicitly says the endpoint must always return the same 200 message and must not leak existence). Wrap these DB operations in try/except (rollback on failure) and always return the generic 200 response regardless of errors.

Example: `try: ... commit() except Exception: await db.rollback()` and still return the generic MessageResponse.